### PR TITLE
Update `docker-compose.yml` to comply with contemporary schema

### PR DESCRIPTION
**Pull request by jarrettj**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

Starting docker fails. Updated docker file. 

Composer expects PHP 7.4 but Dockerfile references 7.3.

### CODE REVIEW

**Comment by Ocramius** [comment]
Oh wow, we should probably have a CI pipeline entry for verifying this: it looks like this is a super-ancient `docker-compose.yml` format.

**Comment by Ocramius** [comment]
@jarrettj are you able to add a `.github/workflows/` entry that does a `docker-compose build` and `docker-compose run ... echo "ok"` perhaps?

**Comment by jarrettj** [comment]
Hey, sure @Ocramius . Will give it a go 👌🏽.

**Comment by jarrettj** [comment]
@Ocramius first time using github workflow. 

How do I get past the DCO fail above? Thanks.

**Comment by jarrettj** [comment]
All good, followed the docs :)

File: .github/workflows/docker-build.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: docker-build
+
+on:
+  push:
```
Comment (Ocramius) [inline .github/workflows/docker-build.yml:4]: Should have `on: pr` too

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-build.yml (hunk 1 above)
Comment (jarrettj) [reply .github/workflows/docker-build.yml:4]: Updated.

Comment (Ocramius) [reply .github/workflows/docker-build.yml:4]: LGTM, thanks! https://github.com/jarrettj/laminas-mvc-skeleton/runs/6225033761?check_suite_focus=true#step:3:755

**Review by Ocramius — APPROVED** [review]

**Pull request merged**